web/src/engine/websites/Digimon.ts

```diff
@@ -0,0 +1,56 @@
+import { Tags } from '../Tags';
+import icon from './Digimon.webp';
+import { type Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS } from '../platform/FetchProvider';
+
+function ChapterScript(mangaId: string): string {
+    return `new Promise(resolve => {
+        resolve([...document.querySelectorAll('div[id="${mangaId}"] li.pg-ep_list__item a')].map(chapter => {
+            return {
+                id: new URL(chapter.pathname, window.location).pathname,
+                title: chapter.text.trim()
+            }
+        }).filter(element => element.id != window.location.pathname));
+    });
+`;
+}
+
+@Common.PagesSinglePageJS(`[...document.querySelectorAll('div.swiper ul li[aria-label] > img')].reverse().map(image => new URL(image.src, window.location).href);`, 500)
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    private readonly mangaIdRegexp = new RegExp(/#([^/]+$)/);
+
+    public constructor() {
+        super('digimon', 'Digimon', 'https://digimon.net', Tags.Media.Manga, Tags.Language.Japanese, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExpSafe(`^${this.URI.origin}/digimoncomic/(en/)?#[^/]+$`).test(url);
```

Comment:
The matcher for the hash in the URL seems to make no sense, should slashes really not be allowed in the hash part of the URL?